core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -40,17 +40,26 @@
   private static final Logger LOG = LoggerFactory.getLogger(ResolvingFileIO.class);
   private static final String DEFAULT_SCHEME = "fs";
   private static final String FALLBACK_IMPL = "org.apache.iceberg.hadoop.HadoopFileIO";
+  private static final String S3_FILE_IO_IMPL = "org.apache.iceberg.aws.s3.S3FileIO";
   private static final Map<String, String> SCHEME_TO_FILE_IO = ImmutableMap.of(
       DEFAULT_SCHEME, FALLBACK_IMPL,
-      "s3", "org.apache.iceberg.aws.s3.S3FileIO",
-      "s3a", "org.apache.iceberg.aws.s3.S3FileIO",
-      "s3n", "org.apache.iceberg.aws.s3.S3FileIO"
+      "s3", S3_FILE_IO_IMPL,
+      "s3a", S3_FILE_IO_IMPL,
+      "s3n", S3_FILE_IO_IMPL
```

Comment:
Nit: This change is good, but we _might_ want to put it in another PR to make it easier for people to cherry-pick only PRs that contain bugs (in my view, the lack of a no-arg constructor in `ResolvingFileIO` is a bug). And then possibly update the title of this PR to make it more clear that this is fixing a bug such as `Core: Add missing no-arg constructor in ResolvingFileIO`.

If other's feel differently, then by all means it can stay in this PR.